pg_search/src/postgres/customscan/explain.rs

```diff
@@ -82,6 +83,115 @@ pub fn format_for_explain<T: Serialize>(value: &T) -> String {
     serde_json::to_string(&json_value).unwrap_or_else(|_| "Error".to_string())
 }
 
+/// Recursively inject estimated_docs from QueryWithEstimates tree into JSON
+///
+/// This function walks both the QueryWithEstimates tree and the corresponding JSON
+/// structure in parallel, injecting `estimated_docs` fields at matching query nodes.
+fn inject_estimates_into_json(
+    json_value: &mut serde_json::Value,
+    estimate_tree: &QueryWithEstimates,
+) {
+    use serde_json::Value;
+
+    // Inject estimate at this node if available
+    if let Some(estimated_docs) = estimate_tree.estimated_docs {
+        if let Value::Object(obj) = json_value {
+            obj.insert(
+                "estimated_docs".to_string(),
+                Value::Number(estimated_docs.into()),
+            );
+        }
+    }
+
+    // Recursively process children based on query structure
+    if let Value::Object(obj) = json_value {
+        // Handle boolean queries (must, should, must_not)
+        if let Some(Value::Object(boolean)) = obj.get_mut("boolean") {
+            inject_into_boolean_query(boolean, estimate_tree);
+        }
+        // Handle with_index wrapper
+        else if let Some(Value::Object(with_index)) = obj.get_mut("with_index") {
+            if let Some(first_child) = estimate_tree.children().first() {
+                if let Some(query) = with_index.get_mut("query") {
+                    inject_estimates_into_json(query, first_child);
+                }
+            }
+        }
+        // Handle score_adjusted
+        else if let Some(Value::Object(score_adjusted)) = obj.get_mut("score_adjusted") {
+            if let Some(first_child) = estimate_tree.children().first() {
+                if let Some(query) = score_adjusted.get_mut("query") {
```

Comment:
Rather than making these conditional with a silent failure, please `expect` or `unwrap` them so that an error message would be rendered if our expectations about the data shape were not met.